misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -0,0 +1,408 @@
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createBuiltinSimpleNameReferenceNode;
+import static io.ballerina.converters.util.ConverterUtils.convertOpenAPITypeToBallerina;
+import static io.ballerina.converters.util.ConverterUtils.escapeIdentifier;
+import static io.ballerina.converters.util.ConverterUtils.extractReferenceType;
+
+/**
+ * Implements functionality to convert Json (object or schema) to a ballerina record.
+ *
+ * The conversion is done by using openAPI schema as an intermediary.
+ */
+public class JsonToRecordConverter {
+    private static final PrintStream outStream = System.err;
+    private static List<TypeDefinitionNode> typeDefinitionNodeList;
+
+    /**
+     * This method takes in a json Schema and returns the Ballerina record nodes.
+     * @param jsonSchema json string for the schema
+     * @return {@link ArrayList<TypeDefinitionNode>}
+     * @throws IOException
+     * @throws ConverterException
+     */
+    public static ArrayList<TypeDefinitionNode> fromSchema(String jsonSchema) throws IOException,
+            ConverterException {
+        OpenAPI model = parseJSONSchema(jsonSchema);
+        return generateRecords(model);
+    }
+
+    /**
+     * This method takes in a json object and returns the Ballerina record nodes.
+     *
+     * @param jsonString json object string
+     * @return {@link ArrayList<TypeDefinitionNode>}
+     * @throws ConverterException
+     * @throws IOException
+     */
+    public static ArrayList<TypeDefinitionNode> fromJSON(String jsonString) throws ConverterException, IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        JsonNode inputJsonObject = objectMapper.readTree(jsonString);
+        Map<String, Object> schema = SchemaGenerator.generate(inputJsonObject);
+        String schemaJson = objectMapper.writeValueAsString(schema);
+        OpenAPI model = parseJSONSchema(schemaJson);
+        return generateRecords(model);
+    }
+
+    /**
+     * Generates Ballerina Record nodes for given OpenAPI model.
+     *
+     * @param openApi OpenAPI model
+     * @return {@link ArrayList<TypeDefinitionNode>}  List of Record Nodes
+     * @throws ConverterException in case of bad record fields
+     */
+    public static ArrayList<TypeDefinitionNode> generateRecords(OpenAPI openApi) throws ConverterException {
+        // TypeDefinitionNodes their
+        typeDefinitionNodeList = new LinkedList<>();
+
+        if (openApi.getComponents() != null) {
+            //Create typeDefinitionNode
+            Components components = openApi.getComponents();
+            if (components.getSchemas() != null) {
+                Map<String, Schema> schemas = components.getSchemas();
+                for (Map.Entry<String, Schema> schema: schemas.entrySet()) {
+                    List<String> required = schema.getValue().getRequired();
+
+                    //1.typeKeyWord
```

Comment:
Shall we remove line numbers 119, 121,124,125, 127,129 comments :)

Reply:
Addressed with 5c09fd7